src/handlers/messageHandler.ts

```diff
@@ -1,30 +1,75 @@
-import { type Message, MessageActionRow, MessageButton, NewsChannel, TextChannel } from "discord.js";
-import { getConfig } from "../helpers/configHelpers";
-import { getMessage, resetMessageContext, addMessageContext, isAutoThreadChannel } from "../helpers/messageHelpers";
-import { getRequiredPermissions } from "../helpers/permissionHelpers";
+// ________________________________________________________________________________________________
+//
+// This file is part of Needle.
+//
+// Needle is free software: you can redistribute it and/or modify it under the terms of the GNU
+// Affero General Public License as published by the Free Software Foundation, either version 3 of
+// the License, or (at your option) any later version.
+//
+// Needle is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License along with Needle.
+// If not, see <https://www.gnu.org/licenses/>.
+//
+// ________________________________________________________________________________________________
+
+import { type Message, MessageActionRow, MessageButton, NewsChannel, TextChannel, ThreadChannel } from "discord.js";
+import { emojisEnabled, getConfig, includeBotsForAutothread } from "../helpers/configHelpers";
+import { getMessage, resetMessageContext, addMessageContext, isAutoThreadChannel, getHelpButton, replaceMessageVariables, getThreadAuthor } from "../helpers/messageHelpers";
+import { getRequiredPermissions, getSafeDefaultAutoArchiveDuration } from "../helpers/permissionHelpers";
 
 export async function handleMessageCreate(message: Message): Promise<void> {
-	const clientUser = message.client.user;
+	// Server outage
+	if (!message.guild?.available) return;
+
+	// Not logged in
+	if (message.client.user === null) return;
+
+	if (message.system) return;
+	if (!message.channel.isText()) return;
+	if (!message.inGuild()) return;
+	if (message.author == message.client.user) return;
```

Comment:
Objects are *always* unequal (unless they both have the same reference). Comparing the users' IDs is a better approach.
```suggestion
	if (message.author.id === message.client.user.id) return;
```

Reply:
Fantastic catch, thank you 💜